dll/win32/msafd/misc/dllmain.c

```diff
@@ -1783,6 +1786,12 @@ WSPAccept(
     AcceptSocketInfo->SharedData->State = SocketConnected;
     AcceptSocketInfo->SharedData->ConnectTime = GetCurrentTimeInSeconds();
 
+    /* Inherit non-blocking state to new socket */
+    if (Socket->SharedData->NonBlocking)
+    {
+        AcceptSocketInfo->SharedData->NonBlocking = Socket->SharedData->NonBlocking;
+    }
```

Comment:
You can assign this unconditionally:
```suggestion
    AcceptSocketInfo->SharedData->NonBlocking = Socket->SharedData->NonBlocking;
```